geode-redis/src/main/java/org/apache/geode/redis/internal/executor/cluster/CRC16.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.internal.executor.cluster;
+
+public class CRC16 {
+
+  // CCITT/SDLC/HDLC x^16 + x^12 + x^5 + 1 (CRC-16-CCITT)
+  private static final int CCITT_POLY = 0x8408;
+  private static final short[] crcTable = new short[256];
+
+  // Create the table up front
+  static {
+    int poly = reverseInt16(CCITT_POLY);
+
+    for (int x = 0; x < 256; x++) {
+      int w = x << 8;
+      for (int i = 0; i < 8; i++) {
+        if ((w & 0x8000) != 0) {
+          w = (w << 1) ^ poly;
+        } else {
+          w = w << 1;
+        }
+      }
+      crcTable[x] = (short) w;
+    }
+  }
+
+  /**
+   * Calculate CRC with most significant byte first. Assume all inputs are valid.
+   *
+   * @param data the byte array to use
+   * @param start starting index into the byte array
+   * @param end ending index (exclusive) into the byte array
+   */
+  public static int calculate(byte[] data, int start, int end) {
```

Comment:
Still thinking, but will it be ok to keep the value as short instead of int ? To keep the algo similar to that at https://redis.io/topics/cluster-spec Appendix A

Reply:
Will do.